deployment/registry.py

```diff
@@ -97,20 +102,27 @@ def read_registry(filepath: Path) -> List[RegistryEntry]:
     return registry_entries
 
 
-def write_registry(entries: List[RegistryEntry], filepath: Path) -> Path:
+def write_registry(entries: List[RegistryEntry], filepath: Path, silent: bool = False) -> Path:
     """Writes a nucypher-style contract registry to a file."""
 
     if not entries:
         print("No entries provided.")
         return filepath
 
+    # Sort registry entries to enforce common order
+    # See https://github.com/nucypher/nucypher-contracts/issues/192
+    entries.sort(key=lambda entry: (str(entry.chain_id), entry.name))
+
     data = defaultdict(dict)
     for entry in entries:
+        entry_abi = list(entry.abi)
+        entry_abi.sort(key=lambda d: (d["type"], d.get("name", "")))
+
         data[str(entry.chain_id)][entry.name] = {
             "address": entry.address,
-            "abi": entry.abi,
+            "abi": entry_abi,
             "tx_hash": entry.tx_hash,
-            "block_number": entry.block_number,
+            "block_number": int(entry.block_number),
```

Comment:
👍 - this indirectly takes care of https://github.com/nucypher/nucypher-contracts/issues/194.